app/bundles/LeadBundle/Entity/LeadListRepository.php

```diff
@@ -207,6 +207,36 @@ public function getLeadLists($lead, $forList = false, $singleArrayHydration = fa
     }
 
     /**
+     * Check Lead segments by ids.
+     *
+     * @param Lead $lead
+     * @param $ids
+     *
+     * @return bool
+     */
+    public function checkLeadSegmentsByIds(Lead $lead, $ids)
+    {
+        if (empty($ids)) {
+            return false;
+        }
+
+        $q = $this->_em->getConnection()->createQueryBuilder();
+        $q->select('l.id')
+            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l');
+        $q->join('l', MAUTIC_TABLE_PREFIX.'lead_lists_leads', 'x', 'l.id = x.lead_id')
+            ->where(
+                $q->expr()->andX(
+                    $q->expr()->in('x.leadlist_id', ':ids'),
+                    $q->expr()->eq('l.id', ':leadId')
+                )
+            )
+            ->setParameter('ids', implode(',', $ids))
```

Comment:
An array should be passed here, not imploded array as a string. See http://doctrine-orm.readthedocs.io/projects/doctrine-dbal/en/latest/reference/data-retrieval-and-manipulation.html#list-of-parameters-conversion

The way it would work now is that Doctrine would use only the first value of the array in the IN statement. You can see it yourself when you `var_dump($q->getSql(), $q->getParameters());die;